pkg/controller/registry/resolver/cache.go

```diff
@@ -354,10 +371,15 @@ func (s SortableSnapshots) Less(i, j int) bool {
 		return false
 	}
 
-	// the rest are sorted first in namespace preference order, then by name
+	// the rest are sorted first in namespace preference order, priority, then by name
```

Comment:
I think we want ... > "same namespace as the subscription" > catalog priority > lexical namespace name > ..., to preserve the existing behavior. We definitely don't want lexical namespace name > catalog priority in general, because the motivation for introducing catalog priorities is to make preferences less arcane.